### DISCUSSION TITLE
how to trigger validation(regex) on concat value of the 2 fields(country code+phone number)

### DISCUSSION THREAD

**Discussion by hinsheva** [react-hook-form/react-hook-form#6664] [Q&A] [answered]
**How to trigger validation(regex) on concat value of the 2 fields (country code dropdown+phone number input field)** in the way that user input field(phone number) won't contain the added prefix value after all magic tricks for running validation per regex behind the scenes?

So i have 2 separate fields in my form and I added the logic to concat this values and trigger validation per regex, and looks like the input is valid both on the field level and the form level, so validation works. But I don't know how to hide from user this "full value" in the phoneNumber field so it won't include the prefix too as long as I have separate dropdown for it?

```
const handleOnBlur = ([evt]) => {
  const phone = evt.target.value;
  const removeCountryPrefix = phone && phone.replace(countryPrefix, '');
  const trimmedPhoneNumber = removeCountryPrefix.replace(/[^0-9]/g, '').trim();
  const fullPhoneNumber = countryPrefix && `${countryPrefix + trimmedPhoneNumber}`;
  setValue('phoneNumber', fullPhoneNumber && fullPhoneNumber.replace(/^0*/, ''));
  triggerValidation('phoneNumber');
};
```
_Note: setting back the value of the phoneNumber through 'setValue' actually helps to set the value without the country code, **but** when next time the same validation(per regex) gets triggered, the phone number becomes invalid again(as it's missing the country code part now as regex requires)._

<img width="403" alt="Screenshot 2021-09-28 at 18 05 24" src="https://user-images.githubusercontent.com/38138872/135125137-ee33b47f-7f01-4653-8085-1663ad8fc5da.png">

Please help...can't solve it. I was hoping maybe there is any feature that exists but i couldn't find it or maybe somebody had similar issue? I'd very much appreciate any help.

Thank you guys a lot in advance :)

**Comment by bluebill1049** [comment, marked as answer]
```tsx
validate: (value) => {
  const areaCode = getValues().areaCode;
  
  // value + areaCode  do your validation and vice verse
}
```

**Reply by hinsheva** [reply]
Hi @bluebill1049, thank you so much for your answer, doing smth similar now and seems it's working :)

**Reply by bluebill1049** [reply]
Awesome to hear!
